From 62196e2b269159a5465f5b8d0ed7cf6f29c3282a Mon Sep 17 00:00:00 2001
From: TristanInSec <tristan.mtn@gmail.com>
Date: Mon, 18 May 2026 11:57:21 -0400
Subject: [PATCH] Avoid null dereference in krb5_rd_cred()

A malformed KRB-CRED message may have fewer entries in the encrypted
ticket-info list than it has in the tickets list.  In each loop
iteration, check if we unexpectedly reached the end of the ticket-info
list to avoid a null dereference.

[ghudson@mit.edu: added comment; rewrote commit message]

(cherry picked from commit 48afa9abb89ab2176bb20624d87d010b9984fc08)

ticket: 9211
version_fixed: 1.22.3
---
 src/lib/krb5/krb/rd_cred.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lib/krb5/krb/rd_cred.c b/src/lib/krb5/krb/rd_cred.c
index d89a98ae0fd..c341eba153d 100644
--- a/src/lib/krb5/krb/rd_cred.c
+++ b/src/lib/krb5/krb/rd_cred.c
@@ -98,6 +98,11 @@ make_cred_list(krb5_context context, krb5_cred *krbcred,
             goto cleanup;
 
         info = encpart->ticket_info[i];
+        if (info == NULL) {
+            /* We unexpectedly reached the end of the encrypted ticket info. */
+            ret = KRB5KRB_AP_ERR_MODIFIED;
+            goto cleanup;
+        }
         ret = krb5_copy_principal(context, info->client, &list[i]->client);
         if (ret)
             goto cleanup;
